grafana/resource_team.go

```diff
@@ -0,0 +1,264 @@
+package grafana
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+type TeamMember struct {
+	ID    int64
+	Email string
+}
+
+type MemberChange struct {
+	Type   ChangeMemberType
+	Member TeamMember
+}
+
+type ChangeMemberType int8
+
+const (
+	AddMember ChangeMemberType = iota
+	RemoveMember
+)
+
+func ResourceTeam() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateTeam,
+		Read:   ReadTeam,
+		Update: UpdateTeam,
+		Delete: DeleteTeam,
+		Exists: ExistsTeam,
+		Importer: &schema.ResourceImporter{
+			State: ImportTeam,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"team_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"members": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func CreateTeam(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gapi.Client)
+	name := d.Get("name").(string)
+	email := d.Get("email").(string)
+	teamID, err := client.AddTeam(name, email)
+	if err != nil && err.Error() == "409 Conflict" {
+		return errors.New(fmt.Sprintf("Error: A Grafana Team with the name '%s' already exists.", name))
+	}
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strconv.FormatInt(teamID, 10))
+	return UpdateMembers(d, meta)
+}
+
+func ReadTeam(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gapi.Client)
+	teamID, _ := strconv.ParseInt(d.Id(), 10, 64)
+	resp, err := client.Team(teamID)
+	if err != nil && err.Error() == "404 Not Found" {
```

Comment:
```suggestion
	if err != nil && strings.HasPrefix(err.Error(), "status: 404") {
```

Full error string would be `status: 404, body: {"message":"Team not found"}` because of https://github.com/grafana/grafana-api-golang-client/blob/c87fc20af1eab8c2dce94aa3e07a631508150ce5/client.go#L69. It's unfortunate that we're matching a string pattern here, but it'll have to do for now.